apps/src/organized_segmentation_demo.cpp

```diff
@@ -54,9 +54,10 @@ displayEuclideanClusters (const pcl::PointCloud<PointT>::CloudVectorType &cluste
   for (size_t i = 0; i < clusters.size (); i++)
   {
     sprintf (name, "cluster_%d" , int (i));
-    pcl::visualization::PointCloudColorHandlerCustom<PointT> color0(boost::make_shared<pcl::PointCloud<PointT> >(clusters[i]),red[i%6],grn[i%6],blu[i%6]);
-    if (!viewer->updatePointCloud (boost::make_shared<pcl::PointCloud<PointT> >(clusters[i]),color0,name))
-      viewer->addPointCloud (boost::make_shared<pcl::PointCloud<PointT> >(clusters[i]),color0,name);
+    pcl::PointCloud<PointT>::Ptr cluster_cloud (new pcl::PointCloud<PointT> (clusters[i]));
+    pcl::visualization::PointCloudColorHandlerCustom<PointT> color0(cluster_cloud,red[i%6],grn[i%6],blu[i%6]);
+    if (!viewer->updatePointCloud (cluster_cloud,color0,name))
+      viewer->addPointCloud (cluster_cloud,color0,name);
```

Comment:
The prior behavior was to generate shared_pointers and invoke separate copy ctors i.e., every time make_shared is invoked, a copy is being created. 

With the new behavior, the copy ctor is invoked only once and then you simply pass the same shared pointer around. This is not a `const` shared pointer so I'm slightly concerned these methods might have the freedom to modify the contents. Hopefully, if the interface was done right, we can quickly test this by setting
```cpp
pcl::PointCloud<PointT>::ConstPtr cluster_cloud (new pcl::PointCloud<PointT> (clusters[i]));
```
and see if it still compiles.

Reply:
Both `updatePointCloud` and `addPointCloud` take `ConstPtr`, so they don't modify anything. And yeah, it still compiles with your suggestion. Do you want me to commit this change?

Reply:
Committed anyway